hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/alias/CredentialProviderFactory.java

```diff
@@ -110,4 +110,31 @@ public static List<CredentialProvider> getProviders(Configuration conf
     }
     return result;
   }
+
+  /**
+   * Get CredentialProvider for hive provider path.
+   *
+   * @param conf configuration object
+   * @param credProviderPath provider path
+   * @return credProviderPath object
+   */
+  public static CredentialProvider getProvider(Configuration conf,
+      String credProviderPath) throws IOException {
```

Comment:
have caller pass in the URI; makes it their problem to create that from a Path, string, etc.